client/packages/common/src/authentication/api/hooks/usePreferences.ts

```diff
@@ -13,17 +13,37 @@ type PreferencesQuery = {
   preferences: PreferencesNode;
 };
 
+// Map of complex preference fields that require subfield selection,
+// If they are a complex type, they must be included in the query or we get an error from the graphql server
+// These must be kept in sync with the GraphQL schema.
+// If a new complex preference type is added, add it here with its subfields.
+const COMPLEX_PREFERENCE_FIELDS: Record<string, string> = {
+  warnWhenMissingRecentStocktake: `
+    enabled
+    maxAge
+    minItems
+  `,
+};
+
 /** Fields undefined until query has loaded */
 export const usePreferences = (): Partial<PreferencesNode> => {
   const { client } = useGql();
   const { storeId } = useAuthContext();
 
+  // Build field list with subfields for complex types
+  const fields = Object.values(PreferenceKey)
+    .map(key => {
+      const subfields = COMPLEX_PREFERENCE_FIELDS[key];
+      return subfields ? `${key} { ${subfields} }` : key;
+    })
+    .join('\n');
```

Comment:
Ah of course, hadn't considered what pain this might cause for GQL. An alternative would be having this discoverable from the API through gql introspection? Makes an extra round trip for displaying the UI though (if we're not already doing that, I haven't looked at this part deeply).